### DISCUSSION TITLE
Decimal and float handled differently

### DISCUSSION THREAD

**Discussion by hongyuan1306** [fastapi/fastapi#10403] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from decimal import Decimal

from fastapi import FastAPI
from fastapi.testclient import TestClient
from pydantic import BaseModel

app = FastAPI()
client = TestClient(app)


class ResponseModel(BaseModel):
    amount: Decimal
    paid: float


@app.get('/path', response_model=ResponseModel)
async def func():
    return {'amount': '1.05', 'paid': '1.05'}


response = client.get('/path')
print(response.json())
```

### Description

According to fastapi doc under https://fastapi.tiangolo.com/tutorial/extra-data-types/, `Decimal` type shall be handled the same as `float` both in request and response.

However, in a json resopnse body, `Decimal` is serialized as a string, while a float as a number. Thus running the example code will give:

```
{'amount': '1.05', 'paid': 1.05}
```

How can I enforce a serialization of all `Decimal` as number in fastapi without changing each and every model in which a decimal field is used?

### Operating System

Linux

### FastAPI Version

0.103.0

### Pydantic Version

2.4.2

### Python Version

Python 3.11.6

**Comment by SCSR-is-too-short-username** [comment]
To my understanding decimals in python should be defined as `Decimal('1.05')` and not as `'1.05'` as you do in the return statement (but it probably doesn't matter here since it seems to work fine on that part).

The most likely reason why the result is given as a string is because decimals are more precise that floats. If you were to convert the decimal to a number (float), you would lose the precision advantage from using decimal in the first place.

**Comment by hongyuan1306** [comment]
I can understand why pydantic v2 serialize python `Decimal` as a json string in stead of number.

Pydantic v1 serializes `Decimal` as a float and the fast documentation was correct then. Now as fastapi migrates to pydantic v2 as default, the documentation needs an update.
